### PR TITLE
run engadmin migrations and get setup for running side car container in ecs

### PR DESCRIPTION
**Pull request by rdhariwal**
## Description

This PR handles the build, tag and registering of eng admin migrations container along with running it. This PR also sets up the ground work to deploy engadmin app (task_def work that can handle sidecar container)

**Important: This PR depends on the following two PR's:**
1. [infra PR](https://github.com/transcom/ppp-infra/pull/898)
1. [milmove_admin PR](https://github.com/transcom/milmove_admin/pull/19)

## Reviewer Notes

Watch out for hidden deps that I might have over looked are only available in app-experimental env

## Code Review Verification Steps

* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/tree/master/docs/database.md#zero-downtime-migrations))
  * [ ] Have been communicated to #dp3-engineering
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-407) for this change

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -201,6 +219,38 @@ commands:
...
+          name: Get Digest from filesystem
+          command: echo 'export ECR_DIGEST_NGINX=$(cat /home/circleci/transcom/mymove/bin/sha/ECR_DIGEST_app-engadmin-nginx)' | tee -a "${BASH_ENV}"
+      - deploy:
+          name: Deploy app service
+          command: scripts/do-exclusively --job-name ${CIRCLE_JOB} scripts/ecs-deploy-service-multi-containers app-engadmin "${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/app-engadmin@${ECR_DIGEST}" "${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/app-engadmin-nginx@${ECR_DIGEST_NGINX}" "${APP_ENVIRONMENT}"
+          no_output_timeout: 20m
+      - run:
+          name: Health Check
+          command: bin/health-checker --schemes http,https --hosts << parameters.health_check_hosts >> --tries 10 --backoff 3 --log-level info --timeout 15m
+      - announce_failure
```
Comment (rdhariwal) [inline .circleci/config.yml:252]: this can be moved to a sep branch

File: .circleci/config.yml [hunk 2]
```diff
@@ -749,8 +799,18 @@ jobs:
             MIGRATION_PATH: 'file:///home/circleci/transcom/mymove/migrations/app/schema;file:///home/circleci/transcom/mymove/migrations/app/secure'
       - run:
           name: clone mymove_admin repo
-          command: git clone https://github.com/transcom/milmove_admin.git
+          command: git clone -b rd-enable-iam-auth --single-branch https://github.com/transcom/milmove_admin.git
```
Comment (rdhariwal) [inline .circleci/config.yml:802]: will have to merge https://github.com/transcom/milmove_admin/pull/19 before switching back to master branch

File: cmd/ecs-deploy/task_def.go [hunk 3]
```diff
@@ -595,39 +665,47 @@ func taskDefFunction(cmd *cobra.Command, args []string) error {
...
 	}
 
-	// Registration is never allowed by default and requires a flag
-	if v.GetBool(dryRunFlag) {
-		// Format the new task def as JSON for viewing
-		newTaskDefJSON, jsonErr := jsonutil.BuildJSON(newTaskDefInput)
-		if jsonErr != nil {
-			quit(logger, nil, err)
+	//if sidecar container is present then add it to container definition
+	if len(sidecarImageURI) > 0 {
```
Comment (rdhariwal) [inline cmd/ecs-deploy/task_def.go:683]: this code also belongs in a sep branch but will be no op and is safe to merge

Comment (rdhariwal) [reply cmd/ecs-deploy/task_def.go:683]: sep branch = when eng admin gets deployed.. this code supports sidecar image

File: .circleci/config.yml (hunk 1 above)
Comment (chrisgilmerproj) [reply .circleci/config.yml:252]: Yeah, let's move the app deploy stuff to another branch.

File: .circleci/config.yml [hunk 4]
```diff
@@ -749,8 +799,18 @@ jobs:
...
           working_directory: ~/transcom
+      - run:
+          name: install deps
+          command: pip3 install -r app/requirements.txt
```
Comment (chrisgilmerproj) [inline .circleci/config.yml:806]: I don't think python2 is installed on this container so you can use `pip`. I bet `pip3` is just a symlink to `pip`.

File: cmd/ecs-deploy/task_def.go [hunk 5]
```diff
@@ -222,6 +228,7 @@ func initTaskDefFlags(flag *pflag.FlagSet) {
 	flag.String(serviceFlag, "app", fmt.Sprintf("The service name (choose %q)", services))
 	flag.String(environmentFlag, "", fmt.Sprintf("The environment name (choose %q)", environments))
 	flag.String(imageURIFlag, "", "The URI of the container image to use in the task definition")
+	flag.String(sidecarImageURIFlag, "", "The URI of the sidecar container image to use in the task definition")
```
Comment (chrisgilmerproj) [inline cmd/ecs-deploy/task_def.go:231]: In my ideal world you would just specify multiple `--image-uri` flags and they would create a list of images to add. Instead of having a separate flag here.

File: cmd/ecs-deploy/task_def.go [hunk 6]
```diff
@@ -506,8 +515,17 @@ func taskDefFunction(cmd *cobra.Command, args []string) error {
 	// Entrypoint
 	entryPoint := v.GetString(entryPointFlag)
 	entryPointList := strings.Split(entryPoint, " ")
-	commandName := entryPointList[0]
-	subCommandName := entryPointList[1]
+	var commandName, subCommandName string
```
Comment (chrisgilmerproj) [inline cmd/ecs-deploy/task_def.go:518]: Update the comment here to explain what is going on here. Unless you know what commands might show up this could be an issue when developing.

File: .circleci/config.yml [hunk 7]
```diff
@@ -749,8 +799,18 @@ jobs:
...
+          working_directory: ~/transcom/milmove_admin
+      - run:
+          name: generate key
+          command: make prepare_key_standalone
+          working_directory: ~/transcom/milmove_admin
+          environment:
+            LOGIN_GOV_JWK_SET_FILENAME: 'keyset.jwk'
```
Comment (mr337) [inline .circleci/config.yml:813]: I don't believe this is needed anymore with the move to circleci var.

File: config/env/experimental.app-engadmin.env [hunk 8]
```diff
@@ -0,0 +1,9 @@
+DB_IAM=true
```
Comment (mr337) [inline config/env/experimental.app-engadmin.env:1]: For experimental does this still hold true?

File: cmd/ecs-deploy/task_def.go [hunk 9]
```diff
@@ -535,6 +553,16 @@ func taskDefFunction(cmd *cobra.Command, args []string) error {
 		if listTargetsByRuleErr != nil {
 			quit(logger, nil, fmt.Errorf("error retrieving targets for rule %q: %w", ruleName, listTargetsByRuleErr))
 		}
+	} else if subCommandName == "migrate" && commandName == "python /home/app/web/manage.py" { //engadmin migrations has its own task def unlike app-migrations below
+		awsLogsStreamPrefix = serviceName
+		awsLogsGroup = fmt.Sprintf("ecs-tasks-app-engadmin-%s", environmentName)
+		containerDefName = fmt.Sprintf("%s-%s", serviceName, environmentName)
+		executionRoleArn = fmt.Sprintf("ecs-task-execution-role-app-engadmin-%s", environmentName)
+		taskRoleArn = fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
```
Comment (chrisgilmerproj) [inline cmd/ecs-deploy/task_def.go:561]: How do we differ from these lines? https://github.com/transcom/mymove/pull/3853/files#diff-2cdb877872b47a22a390bcc6bafb6d3fR531-R533

And the follow up is why are we creating a difference here? My original hope was to modify everything to use the standard naming conventions instead of using bespoke naming conventions. In particular, the reason the `migrate` command is called out is simply because the task role is missing an `s` character, something I ought to be able to fix in terraform. The other reason is because the execution role isn't split out properly and I had to override.

Long story short: names should be defined by this script in the defaults and we translate that to the terraform code instead of trying to make this script conform to what we have in terraform.

File: cmd/ecs-deploy/task_def.go [hunk 10]
```diff
@@ -486,11 +493,13 @@ func taskDefFunction(cmd *cobra.Command, args []string) error {
 	environmentName := v.GetString(environmentFlag)
 	serviceName := v.GetString(serviceFlag)
 	imageURI := v.GetString(imageURIFlag)
+	sidecarImageURI := v.GetString(sidecarImageURIFlag)
 	variablesFile := v.GetString(variablesFileFlag)
 
 	// Short service name needed for RDS, CloudWatch Logs, and SSM
 	serviceNameParts := strings.Split(serviceName, "-")
 	serviceNameShort := serviceNameParts[0]
+	ssmNameSpace := serviceNameShort
```
Comment (chrisgilmerproj) [inline cmd/ecs-deploy/task_def.go:502]: Add this:

```suggestion
	// TODO: SSM should have a namespace for every app which means splitting app and app-client-tls to be more like app-engadmin. This is future work.
	ssmNameSpace := serviceNameShort
```

File: cmd/ecs-deploy/task_def.go (hunk 3 above)
Comment (chrisgilmerproj) [reply cmd/ecs-deploy/task_def.go:683]: I feel like keep it here. Otherwise you may end up causing issues between the branches. Check my comment about using multiple image uris.

File: .circleci/config.yml (hunk 2 above)
Comment (rdhariwal) [reply .circleci/config.yml:802]: it doesnt yet.. but it has some other changes like the key generation logic and iam auth is controlled by flag  so we can merge the milmove_admin pr without side affects

File: config/env/experimental.app-engadmin.env (hunk 8 above)
Comment (chrisgilmerproj) [inline config/env/experimental.app-engadmin.env:1]: @rdhariwal - make sure that in the Django app we are using `DB_IAM` instead of `DB_IAM_AUTH`. We want them to be the same.

File: config/env/experimental.engadmin.migrations.env [hunk 11]
```diff
@@ -0,0 +1,6 @@
+DB_NAME=app
+DB_PORT=5432
+LOGIN_GOV_HOSTNAME=idp.int.identitysandbox.gov
+LOGIN_GOV_CALLBACK_PROTOCOL=https
+LOCAL_HOST_NAME=engadmin.experimental.move.mil
+LOGIN_GOV_CALLBACK_PORT=3000
```
Comment (chrisgilmerproj) [inline config/env/experimental.engadmin.migrations.env:6]: I don't believe that migrations should require the `LOGIN_GOV` values.

File: config/env/prod.app-engadmin.env [hunk 12]
```diff
@@ -0,0 +1,9 @@
+DB_IAM=true
+DB_NAME=app
+DB_PORT=5432
+DB_SSL_MODE=verify-full
+DB_SSL_ROOT_CERT=/bin/rds-ca-2019-root.pem
+DB_USER=ecs_user
+DOMAIN=engadmin.move.mil
+LOG_TASK_METADATA=true
+TLS_ENABLED=true
```
Comment (chrisgilmerproj) [inline config/env/prod.app-engadmin.env:9]: For now delete this file.

File: config/env/staging.app-engadmin.env [hunk 13]
```diff
@@ -0,0 +1,9 @@
+DB_IAM=true
+DB_NAME=app
+DB_PORT=5432
+DB_SSL_MODE=verify-full
+DB_SSL_ROOT_CERT=/bin/rds-ca-2019-root.pem
+DB_USER=ecs_user
+DOMAIN=staging.engadmin.move.mil
+LOG_TASK_METADATA=true
+TLS_ENABLED=true
```
Comment (chrisgilmerproj) [inline config/env/staging.app-engadmin.env:9]: For now delete this file.

File: .circleci/config.yml (hunk 7 above)
Comment (rdhariwal) [reply .circleci/config.yml:813]: so this env is not in circleci.. because its not a "secret" i am inclined to keep it here for clarity but I am open to ideas/arguments if we should move it :)

File: cmd/ecs-deploy/task_def.go (hunk 5 above)
Comment (rdhariwal) [reply cmd/ecs-deploy/task_def.go:231]: > In my ideal world you would just specify multiple `--image-uri` flags and they would create a list of images to add. Instead of having a separate flag here.

hmm.. but how do you figure out what image to route traffic aka which one is sidecar and  what images should are just apps and use nginx for reverse proxy? @chrisgilmerproj

**Comment by mr337** [comment]
Wow look at dangerjs boot being so opinionated

File: .circleci/config.yml (hunk 1 above)
Comment (rdhariwal) [reply .circleci/config.yml]: done, i have saved the file along with a link to clean up commit: https://github.com/transcom/mymove/pull/3853/commits/058791e3d84bcc67f05028b50842fd9fd8ce3dda

File: config/env/experimental.engadmin.migrations.env (hunk 11 above)
Comment (rdhariwal) [reply config/env/experimental.engadmin.migrations.env:5]: image
looks like the env var is part of migration step. I can clean up up in next few prs, leaving it in for now.

File: cmd/ecs-deploy/task_def.go (hunk 5 above)
Comment (rdhariwal) [reply cmd/ecs-deploy/task_def.go:231]: in slack thread we said use fifo, first in first out aka first container should have port mappings.

File: cmd/ecs-deploy/task_def.go (hunk 6 above)
Comment (rdhariwal) [reply cmd/ecs-deploy/task_def.go:522]: done

**Comment by rdhariwal** [comment]
> Two things I want to see before we merge this:
> 
> 1. Use multiple `--image-uri` flags instead of adding the new `--sidecar-image` flag.
> 2. A new script named `scripts/deploy-app-engadmin-migrations` which should mimic `scripts/depoy-app-migrations`. This will give me the power to run this script from my computer and test out everything without deploying via CircleCI.

I believe both asks are done, my last build was with the task_def changes in place deployed successfully so atleast single container deploys are working no issues for side car i will find and refine once we play the deploy engadmin app story @chrisgilmerproj

File: .circleci/config.yml (hunk 7 above)
Comment (chrisgilmerproj) [reply .circleci/config.yml:778]: I think @mr337 was saying that he's got code that eliminates the need for using a file all together. That instead we can just generate env vars and use those instead.

File: cmd/ecs-deploy/task_def.go [hunk 14]
```diff
@@ -221,7 +226,7 @@ func initTaskDefFlags(flag *pflag.FlagSet) {
 	// Task Definition Settings
 	flag.String(serviceFlag, "app", fmt.Sprintf("The service name (choose %q)", services))
 	flag.String(environmentFlag, "", fmt.Sprintf("The environment name (choose %q)", environments))
-	flag.String(imageURIFlag, "", "The URI of the container image to use in the task definition")
+	flag.String(imageURIFlag, "", "The URI of the container(s) image(s) to use in the task definition")
```
Comment (chrisgilmerproj) [inline cmd/ecs-deploy/task_def.go:229]: It's not multiple containers, just multiple container images. The images themselves turn into multiple containers. So maybe taking out the word `container` would clear things up.

```suggestion
	flag.String(imageURIFlag, "", "The URI of the image(s) to use in the task definition")
```

File: cmd/ecs-deploy/task_def.go [hunk 15]
```diff
@@ -485,17 +490,21 @@ func taskDefFunction(cmd *cobra.Command, args []string) error {
 	awsRegion := v.GetString(cli.AWSRegionFlag)
 	environmentName := v.GetString(environmentFlag)
 	serviceName := v.GetString(serviceFlag)
-	imageURI := v.GetString(imageURIFlag)
+	// image uri flag can hold one or more comma separated images, assume first image is the main image
+	imageURIs := v.GetString(imageURIFlag)
+	imageList := strings.Split(imageURIs, ",")
```
Comment (chrisgilmerproj) [inline cmd/ecs-deploy/task_def.go:495]: I was hoping you'd change from `v.GetString()` to `v.GetStringSlice()` so that folks could do `--image-uri URI1 --image-uri URI2` instead of `--image-uri URI1,URI2`. The reasoning being that its easier to visually see the URIs and read them with multiple invocations of the flag instead of multiple, comma delimited URIs.

File: cmd/ecs-deploy/task_def.go (hunk 9 above)
Comment (chrisgilmerproj) [reply cmd/ecs-deploy/task_def.go:564]: @rdhariwal - I think this is the most critical part of the PR you have, which is what things have to differ and which ones don't need to. And then I'd expect some changes back to the ppp-infra IAM roles to fix up the names.

File: scripts/ecs-deploy-service-multi-containers [hunk 16]
```diff
@@ -0,0 +1,121 @@
+#! /usr/bin/env bash
```
Comment (chrisgilmerproj) [inline scripts/ecs-deploy-service-multi-containers:1]: Let's name this `ecs-deploy-app-engadmin` and be specific. A more specific script here will help folks from trying to abstract it too much.

**Comment by rdhariwal** [comment]
> My last comment is really that I want this script to drive the IAM role names. You have a special block of code and in my ideal world we'd delete that and use the defaults generated by this script. And if the script is not providing workable names then I'd like to discuss and possibly update other infrastructure.

Fully agreed on the point, in my ideal world we don't have any `if` condition blocks. I am only doing it because the existing migrations conditional has some very wonky defaults. For example the [taskRoleArn](https://github.com/transcom/mymove/blob/5c5bfc6e3154dd10cddd3dcf3891bdb64677e6bb/cmd/ecs-deploy/task_def.go#L580) for migrations is missing an `s` in its name.  The execution role uses the same role as [app](https://github.com/transcom/mymove/blob/5c5bfc6e3154dd10cddd3dcf3891bdb64677e6bb/cmd/ecs-deploy/task_def.go#L576) 

My proposal is let's add a story to simplify it, fix the roles for migrations in our infra and then let env variables drive them vs the if conditionals we have. Ideally there are no conditionals.. app name and env variables should take care of it all. Thoughts @chrisgilmerproj

File: config/env/experimental.app-engadmin.env (hunk 8 above)
Comment (mr337) [reply config/env/experimental.app-engadmin.env]: Going to resolve this we just want to get this stood up.

**Comment by chrisgilmerproj** [comment]
> Fully agreed on the point, in my ideal world we don't have any `if` condition blocks. I am only doing it because the existing migrations conditional has some very wonky defaults. For example the [taskRoleArn](https://github.com/transcom/mymove/blob/5c5bfc6e3154dd10cddd3dcf3891bdb64677e6bb/cmd/ecs-deploy/task_def.go#L580) for migrations is missing an `s` in its name. The execution role uses the same role as [app](https://github.com/transcom/mymove/blob/5c5bfc6e3154dd10cddd3dcf3891bdb64677e6bb/cmd/ecs-deploy/task_def.go#L576)

I know about these deficiencies, I'm not really looking for you to fix the current stuff just the new stuff for your PR. Here is your code block:

```go
awsLogsStreamPrefix = serviceName
awsLogsGroup = fmt.Sprintf("ecs-tasks-app-engadmin-%s", environmentName)
containerDefName = fmt.Sprintf("%s-%s", serviceName, environmentName)
executionRoleArn = fmt.Sprintf("ecs-task-execution-role-app-engadmin-%s", environmentName)
taskRoleArn = fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
```

Here are the defaults:

```go
executionRoleArn := fmt.Sprintf("ecs-task-execution-role-%s-%s", serviceName, environmentName)
taskRoleArn := fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
family := fmt.Sprintf("%s-%s", serviceName, environmentName)
```

So for instance you have these two lines:

```go
// Your code
taskRoleArn = fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
// default
taskRoleArn := fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
```

They are identical, so why did you repeat the code?

Similarly, for execution, why is `serviceName` not already equivalent to `app-engadmin`?

```go
// Your code
executionRoleArn = fmt.Sprintf("ecs-task-execution-role-app-engadmin-%s", environmentName)
// default
executionRoleArn := fmt.Sprintf("ecs-task-execution-role-%s-%s", serviceName, environmentName)
```

If it was equivalent you could use the default.  See what I'm getting at here? Ideally you'd be able to fall into the `else` block of the if/else chain and just use the defaults without having to make specific `app-engadmin` additions. Because `app-client-tls` can fall into the `else` block so I'd expect `app-engadmin` could as well.

**Comment by rdhariwal** [comment]
> > Fully agreed on the point, in my ideal world we don't have any `if` condition blocks. I am only doing it because the existing migrations conditional has some very wonky defaults. For example the [taskRoleArn](https://github.com/transcom/mymove/blob/5c5bfc6e3154dd10cddd3dcf3891bdb64677e6bb/cmd/ecs-deploy/task_def.go#L580) for migrations is missing an `s` in its name. The execution role uses the same role as [app](https://github.com/transcom/mymove/blob/5c5bfc6e3154dd10cddd3dcf3891bdb64677e6bb/cmd/ecs-deploy/task_def.go#L576)
> 
> I know about these deficiencies, I'm not really looking for you to fix the current stuff just the new stuff for your PR. Here is your code block:
> 
> ```go
> awsLogsStreamPrefix = serviceName
> awsLogsGroup = fmt.Sprintf("ecs-tasks-app-engadmin-%s", environmentName)
> containerDefName = fmt.Sprintf("%s-%s", serviceName, environmentName)
> executionRoleArn = fmt.Sprintf("ecs-task-execution-role-app-engadmin-%s", environmentName)
> taskRoleArn = fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
> ```
> 
> Here are the defaults:
> 
> ```go
> executionRoleArn := fmt.Sprintf("ecs-task-execution-role-%s-%s", serviceName, environmentName)
> taskRoleArn := fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
> family := fmt.Sprintf("%s-%s", serviceName, environmentName)
> ```
> 
> So for instance you have these two lines:
> 
> ```go
> // Your code
> taskRoleArn = fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
> // default
> taskRoleArn := fmt.Sprintf("ecs-task-role-%s-%s", serviceName, environmentName)
> ```
> 
> They are identical, so why did you repeat the code?
> 
> Similarly, for execution, why is `serviceName` not already equivalent to `app-engadmin`?
> 
> ```go
> // Your code
> executionRoleArn = fmt.Sprintf("ecs-task-execution-role-app-engadmin-%s", environmentName)
> // default
> executionRoleArn := fmt.Sprintf("ecs-task-execution-role-%s-%s", serviceName, environmentName)
> ```
> 
> If it was equivalent you could use the default. See what I'm getting at here? Ideally you'd be able to fall into the `else` block of the if/else chain and just use the defaults without having to make specific `app-engadmin` additions. Because `app-client-tls` can fall into the `else` block so I'd expect `app-engadmin` could as well.

ahh I see,  i think [this](https://github.com/transcom/mymove/blob/868b62cae1e4a5dc4ece45b4801857369aa52a4d/cmd/ecs-deploy/task_def.go#L566) is the only reason I need that else if block then, otherwise i can fall back on defaults. I need to override this namespace value so our containers are looking at app-engadmin namespace.  Otherwise defaults can handle everything else.

**Comment by chrisgilmerproj** [comment]
> ahh I see, i think [this](https://github.com/transcom/mymove/blob/868b62cae1e4a5dc4ece45b4801857369aa52a4d/cmd/ecs-deploy/task_def.go#L566) is the only reason I need that else if block then, otherwise i can fall back on defaults. I need to override this namespace value so our containers are looking at app-engadmin namespace. Otherwise defaults can handle everything else.

Great!

File: .circleci/config.yml [hunk 17]
```diff
@@ -81,18 +81,36 @@ commands:
...
           command: scripts/do-exclusively --job-name ${CIRCLE_JOB} scripts/rds-snapshot-app-db $APP_ENVIRONMENT
       - run:
-          name: Run migrations
+          name: Run app migrations
           command: scripts/do-exclusively --job-name ${CIRCLE_JOB} scripts/ecs-run-app-migrations-container ${AWS_ACCOUNT_ID}.dkr.ecr.${AWS_DEFAULT_REGION}.amazonaws.com/app-migrations@${ECR_DIGEST} $APP_ENVIRONMENT
           no_output_timeout: 60m
           environment:
             CHAMBER_RETRIES: 20
       - announce_failure
+  deploy_engadmin_migrations_steps:
```
Comment (chrisgilmerproj) [inline .circleci/config.yml:96]: Add a step to check staging/prod and exit cleanly.

File: config/env/experimental.app-engadmin.env [hunk 18]
```diff
@@ -0,0 +1,4 @@
+DB_NAME=app
+DB_PORT=5432
+DB_USER=ecs_user
+DOMAIN=experimental.engadmin.move.mil
```
Comment (chrisgilmerproj) [inline config/env/experimental.app-engadmin.env:4]:
```suggestion
DOMAIN=engadmin.experimental.move.mil
```

File: config/env/experimental.engadmin.migrations.env [hunk 19]
```diff
@@ -0,0 +1,6 @@
+DB_NAME=app
+DB_PORT=5432
+LOGIN_GOV_CALLBACK_PROTOCOL=https
+LOCAL_HOST_NAME=engadmin.experimental.move.mil
+LOGIN_GOV_CALLBACK_PORT=3000
```
Comment (chrisgilmerproj) [inline config/env/experimental.engadmin.migrations.env:5]:
```suggestion
LOGIN_GOV_CALLBACK_PORT=443
```

Comment (chrisgilmerproj) [reply config/env/experimental.engadmin.migrations.env:5]: Actually, 443 is the default so just remove this line.

File: config/env/experimental.engadmin.migrations.env [hunk 20]
```diff
@@ -0,0 +1,6 @@
+DB_NAME=app
+DB_PORT=5432
+LOGIN_GOV_CALLBACK_PROTOCOL=https
```
Comment (chrisgilmerproj) [inline config/env/experimental.engadmin.migrations.env:3]: This is the default, so remove it.

Comment (chrisgilmerproj) [reply config/env/experimental.engadmin.migrations.env:3]: In this code https://github.com/transcom/milmove_admin/blob/master/app/milmove_admin/settings.py#L201-L265

Use:

```py
os.environ.get("LOGIN_GOV_CALLBACK_PROTOCOL", "https")

File: scripts/deploy-app-engadmin-migrations [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+
+if [ -z "${AWS_ACCOUNT_ID}" ]; then
+  echo "AWS_ACCOUNT_ID not found!"
+  echo "Please add 'AWS_ACCOUNT_ID' to your .envrc.local"
+  exit 1
+fi
+
+APP_ENVIRONMENT="${environment}"
+AWS_DEFAULT_REGION=us-west-2
+CIRCLE_SHA1="dfc873b737939be48245920b12e799d040f4e3b8" #use a static commit until engadmin is deployed
```
Comment (chrisgilmerproj) [inline scripts/deploy-app-engadmin-migrations:26]: Modify this to pull from the health check?

Comment (chrisgilmerproj) [reply scripts/deploy-app-engadmin-migrations:26]: Make a ticket to implement a health check with git-sha for milmove_admin

File: scripts/ecs-deploy-app-engadmin [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+echo "* Registering new task definition"
+set +e
+green_task_def_arn=$("${DIR}/../bin/ecs-deploy" task-def \
+  --aws-account-id "${AWS_ACCOUNT_ID}" \
+  --aws-region us-west-2 \
+  --service "${name}" \
+  --environment "${environment}" \
+  --image "${images}" \
+  --variables-file "${DIR}/../config/env/${environment}.${name}.env" \
+  --entrypoint "milmove_admin.wsgi:application --name milmove_admin --bind 0.0.0.0:8000 --workers 3 --log-level info --error-logfile ''-' --access-logfile '-'" \
```
Comment (chrisgilmerproj) [inline scripts/ecs-deploy-app-engadmin:89]: See if you can't collapse this into the other script, if not its ok.
